itinerant-iterators/app/byteutils.py

```diff
@@ -0,0 +1,42 @@
+from typing import Generator
+
+
+def iter_bits(bytes_: bytes) -> Generator[int, None, None]:
+    """Enumerate over each bit in the given bytes.
+
+    Args:
+        bytes_ (bytes): bytes to enumerate over
+
+    Yields:
+        Generator[int, None, None]: Return each bit as an int (0/1)
+    """
+    for byte in bytes_:
+        bitstring = f"{byte:0>8b}"
+        for bit in bitstring:
+            yield int(bit)
+
+
+def set_bit(value: int, bit: int) -> int:
+    """Set the bit as position on the value.
+
+    Args:
+        value (int): Value to set bit on
+        bit (int): Bit position to set
+
+    Returns:
+        int: value with the bit at position `bit` set
+    """
```

Comment:
This is what I mean by unclear docstrings. It's good that they exist, but it's no good if they do little to help the reader with what the function does. How about this?

```suggestion
    """Set a bit in a byte to one.

    Args:
        value: Byte to set bit on
        bit: Bit index to set

    Returns:
        int: Value with the bit at :bit: location set. 
    """
```